README.md

```diff
@@ -27,4 +27,265 @@
+
+# Instructions for SocketViewer
+`Dockerfile.socket` can be used for easy installation. This chapter provides instructions on building and running examples with SocketViewer support using Docker.
+
+## Building Docker Images
+
+### Docker Image of OpenVSLAM
+
+### Step 1: Clone the required github repository:
+
+If you are going to use non RGBD setup then exceute following in your terminal.
+
+    git clone https://github.com/OpenVSLAM-Community/openvslam_ros.git
+
+If you are going to use RGBD setup then exceute following in your terminal.
+
+    git clone -b wip/approximate-time-rgbd https://github.com/OpenVSLAM-Community/openvslam_ros.git
+
+In case of openvslam using rosbag file, then replace `Dockerfile.socket` with `Dockerfile.socket-rosbag`.
+
+### Step 2: Create docker image
+
+Execute the following commands:
+
+    cd /path/to/openvslam
+    git submodule update -i --recursive
+    docker build -t openvslam-ros-socket -f Dockerfile.socket .
+
+You can accelerate the build of the docker image with `--build-arg NUM_THREADS=<number of parallel builds> ` option in command line argument.
+
+For example:
+
+    # building the docker image with four threads
+    docker build -t openvslam-socket -f Dockerfile.socket . --build-arg NUM_THREADS=4
+
+### Step 3: Docker Image of Server
+Copy viewer folder from the main 'openvslam' [repository](https://github.com/OpenVSLAM-Community/openvslam) to your present working direcory i.e 'openvslam_ros'. Then create docker image for socket server.
+
+Execute the following commands:
+
+    git clone https://github.com/OpenVSLAM-Community/openvslam.git openvslam
+    cp -r openvslam/viewer ./viewer
+    rm -r openvslam/ --force
+    cd viewer
+    docker build -t openvslam-ros-server .
+
+### Step 4: Starting Docker Containers for "openvslam-ros-server"
+
+### On Linux
+Launch the server container and access to it with the web browser in advance. Please specify `--net=host` in order to share the network with the host machine.
+
+Execute the following commands:
+
+`Terminal1:`
+
+    docker run --rm -it --name openvslam-ros-server --net=host openvslam-ros-server
+
+Following message should appear on your terminal.
+
+    WebSocket: listening on *:3000
+    HTTP server: listening on *:3001
+
+After launching, access to `http://localhost:3001/` with the web browser.
+
+### Step 5: Starting Docker Containers for "openvslam-ros-socket"
+
+### Step 5.1: Using USB Camera
+
+Execute the following commands:
+
+
+`Terminal2:`
+Launch the container of OpenVSLAM-socket. The shell interface will be launched in the docker container.Download an ORB vocabulary from GitHub. Publish Images Captured by a USB Camera using package like image_tools.
+
+    xhost +local:
+    docker run --rm -it --name openvslam-ros-socket --device /dev/video0 --net=host openvslam-ros-socket
+    root@hostname:/ros2_ws#curl -sL "https://github.com/OpenVSLAM-Community/FBoW_orb_vocab/raw/main/orb_vocab.fbow" -o orb_vocab.fbow
```

Comment:
I think it would be hard to read without a space after `#`.

Reply:
Updated as per review comment.